types/ip.pp

```diff
@@ -0,0 +1,2 @@
+# https://stackoverflow.com/a/190405/1000630, with minor modifications to syntax
+type Tea::IP = Variant[Tea::IPv4, Tea::IPv6]
```

Comment:
https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/compat/ip_address.pp

Reply:
Sure, adding them to stdlib would make sense. I talked a bit with @igalic about this at cfgmgmtcamp and I think we decided to start here, but having it in stdlib would be better. ...but in that case, why not move all types from tea into stdlib?

Reply:
That's entirely valid and this is partially already the case.